user-registration-app/src/app/views/user-profile-view/user-profile-view.component.ts

```diff
@@ -1,178 +1,135 @@
-import {HttpService} from "../../services/HttpService/HttpService";
+import app from 'firebase/app';
+import { NgProgress } from 'ngx-progressbar';
+import { finalize, switchMap } from 'rxjs/operators';
+import { ToastrService } from 'ngx-toastr';
+import { AuthService } from '../../services/AuthService/auth.service';
+import { RegistrationApiResponse } from '../../models/registration';
+import { HttpService } from '../../services/HttpService/HttpService';
 import { Hackathon } from '../../models/hackathon';
 
 @Component({
   selector: 'app-user-profile-view',
   templateUrl: './user-profile-view.component.html',
   styleUrls: ['./user-profile-view.component.css'],
-  animations: [
-    trigger(
-      'fadeIn', [
-        transition(
-          ':enter',
-          useAnimation(fadeInDown, { params: { timing: 0.25 } })
-        ),
-        transition(
-          ':leave',
-          useAnimation(fadeOutUp, { params: { timing: 0.25 } })
-        )
-      ],
-    )
-  ],
 })
 export class UserProfileViewComponent implements OnInit {
   private static readonly DEFAULT_PROFILE_URL: string = '../../assets/icons/user.png';
-  private emailEditToggled: boolean;
-  private passwordEditToggled: boolean;
-  private nameEditToggled: boolean;
-  public currentPin: string;
-  public i: number;
-  registrations: RegistrationApiResponse[];
   currentHackathon: Hackathon;
+  currentPin = 'No active pin! Don\'t forget to register!';
 
-  constructor(public authService: AuthService, private httpService: HttpService, private progressService: NgProgress, private alertsService: AlertService) {
-    
-  }
+  constructor(
+    private authService: AuthService,
+    private httpService: HttpService,
+    private progressService: NgProgress,
+    private toastrService: ToastrService,
+  ) { }
 
   ngOnInit() {
-    this.emailEditToggled = false;
-    this.passwordEditToggled = false;
-    this.nameEditToggled = false;
     const regObservable = this.httpService.getUserRegistrations()
-    .subscribe(registrations => {
-      this.registrations = registrations;
-      regObservable.unsubscribe();
-      this.getCurrentPin();
-    }, ({ error }) => {
-      if (error.status === 404) {
-        this.alertsService.info('You have not registered for a hackathon yet. We could not find any data for those queries');
-      }
-    });
+      .subscribe(
+        (registrations: RegistrationApiResponse[]) => {
+          registrations.forEach((registration) => {
+            if (registration.hackathon.active) {
+              this.currentPin = registration.pin.toString();
+            }
+          })
+          regObservable.unsubscribe();
+        },
+        ({ error }) => {
+          if (error && error.status === 404) {
+            this.toastrService.info('You have not registered for a hackathon yet. We could not find any data for you');
+          } else {
+            this.toastrService.error('Something has gone terribly wrong');
```

Comment:
lmao

Reply:
I'm not sure about the loading bar thing, but I believe that's because it's trying to query the api server but I switched it over to localhost and forgot to switch it back. There should be something in the console showing this.